### DISCUSSION TITLE
Sending asset-level commands to the wallet (query, transfer, claim)

### DISCUSSION THREAD

**Discussion by michaelpiron** [project-origin/wallet#30] [Ideas]
Based on our interaction with pilot users in the Elia Group ETT implementation, we detected the clear need to be able to send asset-level commands to the wallet/service.
So instead of requesting transfers or claims based on certificate IDs, they need to be able to request transfers/claims based on the asset IDs (the asset that generated the certificates) + walletsection + timeperiod.
It is about adding a layer of higher-level commands:
![image](https://github.com/project-origin/wallet/assets/4469877/fdac1e92-49c1-489b-9a6a-d7da4f0c01ec)

<details>
  <summary>Example API endpoints to illustrate this</summary>
For illustration, here are our current API endpoints, which accommodate this asset-level command need:

1. Querying certificates by asset:
![image](https://github.com/project-origin/wallet/assets/4469877/a7171215-babb-4eed-ad59-aad8a7489516)

2. Transfers by asset:
![image](https://github.com/project-origin/wallet/assets/4469877/4d255c91-e502-4369-9fee-1e3993eb5703)

3. Claims by asset:
![image](https://github.com/project-origin/wallet/assets/4469877/2fbd54fc-5635-4acc-bd17-b577d7e10a86)
</details>

**How can we accommodate this in the Project Origin wallet implementation?**

@wisbech @MartinSchmidt @WouterMarien @torfiriand @Kingaday @Mike4Elia

**Comment by MartinSchmidt** [comment]
My immediate thought is wether or not this belongs in the wallet itself, or in another microservice that interacts with the wallet.

My worry could be that the wallet quickly becomes the new center of everything, and that is definably not what we would want in the long run, how would continually  claiming work, I would rather not implement a rule engine in the wallet itself.

But the asset id could be an attribute of the certificate, and we could possible allow claims between different certs based on an filter, where attribute or ids could be used. That way it would not be a specific thing in the wallet that it only allow it on asset-id but any combination of attributes. 

If you want to continue to support your current API that could then be translated in an intermediate service.
